osbs/utils.py

```diff
@@ -340,9 +349,7 @@ def make_name_from_git(repo, branch, limit=53, separator='-', hash_size=5):
 
     repo = ''.join(repo_chars)
     branch = ''.join(branch_chars)
-    name = separator.join([repo, branch, hash_str])
-
-    return name
+    return separator.join(filter(None, (repo, branch, hash_str)))
```

Comment:
This would mean that repo is branch name has no letters or numbers at all, which I don't think would be allowed, or anyone would make something like this. But even if they did, the TypeError would be an appropriate answer :)